subsys/shell/shell_history.c

```diff
@@ -7,6 +7,13 @@
 #include <shell/shell_history.h>
 #include <string.h>
 
+struct shell_history_item {
+	sys_dnode_t dnode;
+	u16_t len;
+	u16_t padding;
```

Comment:
yes, we need that. it's because the way ring buffer is handled. we want to be able to access all strings stored in the ring buffer without removing them (traversing history). This is done by keeping pointers to strings (stored in the ring buffer) in a list. As long as strings are not chopped we can access them directly by looking into ring buffer memory.

I added description of the concept at the top of `shell_history.c`